src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -546,26 +547,40 @@ void SyncInferRequest::init_tensor(const std::string& name) {
 
             if (!tensor) {
                 ov::Shape tensor_shape;
+                const auto model_prec = port.get_element_type();
                 if (isDynamic) {
-                    const auto model_prec = port.get_element_type();
-                    const auto graph_prec =
-                        output->second->getParentEdgesAtPort(0)[0]->getMemory().getDesc().getPrecision();
-                    OutputControlBlock control_block{model_prec, Shape{shape}};
-
-                    DEBUG_LOG(name,
-                              ", tensor ",
-                              control_block.tensor(),
-                              ", memmngr ",
-                              control_block.tensor()->get_memory()->getMemoryMngr(),
-                              "memory object ",
-                              control_block.tensor()->get_memory().get());
-
-                    tensor = control_block.tensor();
-                    if (model_prec == graph_prec)
-                        m_outputControlBlocks.emplace(std::make_pair(name, std::move(control_block)));
+                    if (model_prec == element::string) {
```

Comment:
It looks like we need to incapsulate this logic into OutputControlBlock ctor to have unified code here. 
Anyway since current solution solves the issue we can refactor in follow-up PR